### PR TITLE
CIRC-734: Splitting out pub/sub registration from main logic.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
Split from https://github.com/folio-org/mod-circulation/pull/556

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/test/java/api/CirculationRulesAPITests.java [hunk 1]
```diff
@@ -422,7 +421,8 @@ public void canReportValidationError() {
 
     JsonObject json = new JsonObject(response.getBody());
 
-    assertThat(json.getString("message"), containsStringIgnoringCase("tab"));
+    assertThat(json.getString("message"),
+      is("Tabulator character is not allowed, use spaces instead."));
```
Comment (alexanderkurash) [inline src/test/java/api/CirculationRulesAPITests.java:425]: I don't remember this in the original PR, though it seems to be benign.

File: pom.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
   <properties>
     <antlr4.version>4.7.2</antlr4.version>
     <drools.version>7.0.0.Final</drools.version>
-    <rmb.version>24.0.0</rmb.version>
+    <rmb.version>30.0.0</rmb.version>
```
[the remark this answers is not in the document]
Comment (bohdan-suprun) [reply pom.xml:28]: The mod-circulation is not an RMB module, the property defines only version of org.folio:utils and org.folio:testing libs.

File: pom.xml [hunk 3]
```diff
@@ -131,12 +141,6 @@
       <version>4.12</version>
       <scope>test</scope>
     </dependency>
-    <dependency>
-      <groupId>org.folio</groupId>
-      <artifactId>util</artifactId>
-      <version>${rmb.version}</version>
-      <scope>test</scope>
-    </dependency>
```
[the remark this answers is not in the document]
Comment (bohdan-suprun) [reply pom.xml:139]: It is moved above on line # 122

File: src/main/java/org/folio/circulation/services/PubSubRegistrationService.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+public class PubSubRegistrationService {
+  private static final Logger logger = LoggerFactory.getLogger(PubSubRegistrationService.class);
+
+  private PubSubRegistrationService() {
+    throw new IllegalStateException();
+  }
+
+  public static CompletableFuture<Boolean> registerModule(Map<String, String> headers,
+                                                          Vertx vertx) {
```
[the remark this answers is not in the document]
Comment (bohdan-suprun) [reply src/main/java/org/folio/circulation/services/PubSubRegistrationService.java:28]: I don't think. I don't know how to apply our standard indention, so probably IDE just reformated it.

File: src/test/java/api/CirculationRulesAPITests.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/test/java/api/CirculationRulesAPITests.java:425]: Seems with the new version of org.folio:testing there is no `containsStringIgnoringCase` method.

File: pom.xml [hunk 5]
```diff
@@ -113,6 +113,16 @@
...
+    <dependency>
+      <groupId>org.folio</groupId>
+      <artifactId>mod-pubsub-client</artifactId>
+      <version>1.1.5</version>
+    </dependency>
+    <dependency>
+      <groupId>org.folio</groupId>
+      <artifactId>util</artifactId>
+      <version>${rmb.version}</version>
+    </dependency>
```
Comment (marcjohnson-kint) [inline pom.xml:125]: This is now being used in production code?

File: pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexanderkurash) [reply pom.xml:139]: Oh, sorry..

File: pom.xml (hunk 5 above)
Comment (alexanderkurash) [reply pom.xml:125]: @marcjohnson-kint Yes, we're using `HttpStatus` enum

File: pom.xml [hunk 6]
```diff
@@ -113,6 +113,16 @@
...
     </dependency>
+    <dependency>
+      <groupId>org.folio</groupId>
+      <artifactId>mod-pubsub-client</artifactId>
+      <version>1.1.5</version>
+    </dependency>
+    <dependency>
+      <groupId>org.folio</groupId>
+      <artifactId>util</artifactId>
+      <version>${rmb.version}</version>
```
Comment (alexanderkurash) [inline pom.xml:124]: Have you checked that it's not getting in the way of PubSub client dependency and the client doesn't throw NoClassDefFoundError when it tries to register or send events?

File: src/test/java/api/CirculationRulesAPITests.java (hunk 1 above)
Comment (alexanderkurash) [reply src/test/java/api/CirculationRulesAPITests.java:425]: Got it

File: pom.xml (hunk 6 above)
Comment (alexanderkurash) [reply pom.xml:124]: No, of course it doesn't if tests are passing.. Never mind

File: pom.xml (hunk 5 above)
Comment (marcjohnson-kint) [reply pom.xml:125]: Hmm, ok, it seems a bit of a shame to include a whole new library dependency into the production artefacts for only a status enum

**Pull request merged**